packages/office-ui-fabric-react/src/components/Button/Button.Props.ts

```diff
@@ -44,15 +45,23 @@ export interface IButtonProps extends React.HTMLProps<HTMLButtonElement | HTMLAn
   text?: string;
 
   /**
+  *
+  */
+  contextualMenuItems?: IContextualMenuItem[];
+
+  /**
    * The button icon shown in command or hero type.
-   * Deprecated at v1.2.3, to be removed at >= v2.0.0. Use IconButton instead.
    */
   icon?: string;
 
   /**
+   * The button icon shown to the right of the text
+   */
+  splitIcon?: string;
```

Comment:
I also think you're missing `onRenderMenu`. If I have a very custom "Menu", it would be nice to override.